### ISSUE TITLE
--write=none is not respected

### ISSUE THREAD

**Issue by emulanob** [ansible/ansible-lint#3420]
# Issue Type

- Bug report

# Ansible and Ansible Lint details

```
$ ansible --version                                                   
ansible [core 2.14.5]
  config file = /Users/me/projects/myrepo/ansible.cfg
  configured module search path = ['/Users/me/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/me/Library/Python/3.9/lib/python/site-packages/ansible
  ansible collection location = /Users/me/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.12 (main, Mar 26 2022, 15:51:15) [Clang 13.1.6 (clang-1316.0.21.2)] (/usr/local/opt/python@3.9/bin/python3.9)
  jinja version = 3.1.2
  libyaml = True

$ ansible-lint --version                                              
ansible-lint 6.15.0 using ansible 2.14.5
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behavior

ansible-lint should not modify files when either `write_list: ["none"]` is present in `.ansible-lint` or `--write=none` is passed in the command line.

# Actual Behavior (Bug report only)

The file is modified with all sorts of fixes it is able to perform.

```
$  ansible-lint --write=none myplaybook.yml

[... several errors and warnings are listed according to my settings...]

Read documentation for instructions on how to ignore specific rule violations.

Modified 13 files.
                         Rule Violation Summary                         
 count tag                           profile rule associated tags       
     2 deprecated-module             basic   deprecations               
     1 jinja[spacing]                basic   formatting                 
     2 name[missing]                 basic   idiom                      
     1 name[play]                    basic   idiom (warning)            
     1 yaml[colons]                  basic   formatting, yaml           
     3 yaml[new-line-at-end-of-file] basic   formatting, yaml           
     7 yaml[octal-values]            basic   formatting, yaml           
    13 yaml[truthy]                  basic   formatting, yaml (warning) 
     1 risky-file-permissions        safety  unpredictability           

Failed after min profile: 17 failure(s), 14 warning(s) on 26 files.
```
PS: I already had `write_list: ["none"]` in my .ansible-lint file when I first noticed the issue, so I tried passing `--write=none` in the command line but that result is the same. Also regardless of the file type (playbook/vars/meta/etc).

Here's my `.ansible-lint` file to help debug. 
```
use_default_rules: true

skip_list:
  - fqcn
  - name[casing]
  - role-name
  - yaml[line-length]

# Ansible-lint does not automatically load rules that have the 'opt-in' tag.
# You must enable opt-in rules by listing each rule 'id' below.
enable_list:
  - args
  - empty-string-compare # opt-in
  - no-log-password # opt-in
  - no-same-owner # opt-in
  - yaml

# Ansible-lint does not fail on warnings from the rules or tags listed below
warn_list:
  - name[play]
  - yaml[truthy]

# Some rules can transform files to fix (or make it easier to fix) identified
# errors. `ansible-lint --write` will reformat YAML files and run these transforms.
# By default it will run all transforms (effectively `write_list: ["all"]`).
# You can disable running transforms by setting `write_list: ["none"]`.
# Or only enable a subset of rule transforms by listing rules/tags here.
write_list: ["none"]

# Offline mode disables installation of requirements.yml and schema refreshing
offline: true

# Define required Ansible's variables to satisfy syntax check
extra_vars:
  ansible_environment: lint

# List of additional kind:pattern to be added at the top of the default
# match list, first match determines the file kind.
kinds:
  - playbook: "**/examples/*.{yml,yaml}"
  - tasks: "**/tasks/*.yml"
  - vars: "**/vars/*.yml"
  - vars: "**/vars/**/*.yml"
  - meta: "**/meta/main.yml"
```

**Comment by MarkusTeufelberger** [comment]
Which fixes are actually applied? As far as I understand the feature, YAML rewriting (e.g. turning `yes` into `true`) will always happen, `--write=none` just disables all additional transformations that rules might implement (e.g. writing a FQCN instead of a short module name).

**Comment by chriscroome** [comment]
The [usage docs have](https://ansible-lint.readthedocs.io/usage/):

> '--write=none' resets the list of transforms to allow reformatting YAML without running any of the transforms (ie '--write=none,rule-id' will ignore write_list in the config file and only run the rule-id transform).

So given a `test.yml` file containing:

```yaml
---
- name: Test
  debug:
    var: foo
```

And the command:

```bash
ansible-lint --write=none,fqcn test.yml
```

I would expect that `debug:` would be changed to `ansible.builtin.debug:` however nothing is changed and the result is:

```
Passed with production profile: 0 failure(s), 0 warning(s) on 1 files.
```

But perhaps I'm misunderstanding how this feature is supposed to work?

**Comment by emulanob** [comment]
@MarkusTeufelberger @chriscroome thanks for the clarification, folks. I guess I misinterpreted the feature then. The changes I'm seeing are compatible with YAML formatting indeed (e.g. `---` at the beginning of a file, `yes` into `true`, double quotes around octal permissions, line breaks after the recommended line length...). 

But then there were also some changes to character escaping in complex strings, and those I could not stop to validate at the time, and therefore would prefer to stop from happening
